File: java/src/main/java/vsdk/toolkit/app/options/BatchOptions.java
package vsdk.toolkit.app.options;

public class BatchOptions {
    public boolean exportBinary;
    public String binaryOutputFilename;
    public boolean importBinary;
    public String binaryInputFilename;
    public int iterations; // Radiance method iterations
    public String radianceImageFileNameFormat;
    public String radianceModelFileNameFormat;
    public int saveModulo; // Every n-th iteration, surface model and image will be saved
    public String raytracingImageFileName;
    public int timings = 0;

    public BatchOptions() {
        exportBinary = false;
        binaryOutputFilename = "";
        importBinary = false;
        binaryInputFilename = "";
        iterations = 1;
        radianceImageFileNameFormat = "";
        radianceModelFileNameFormat = "";
        saveModulo = 10;
        raytracingImageFileName = "";
        timings = 0;
    }
}

File: typescript/src/main/vsdk/toolkit/app/options/BatchOptions.ts
export class BatchOptions {
  public exportBinary: boolean;
  public binaryOutputFilename: string;
  public importBinary: boolean;
  public binaryInputFilename: string;
  public iterations: number;
  public radianceImageFileNameFormat: string;
  public radianceModelFileNameFormat: string;
  public saveModulo: number;
  public raytracingImageFileName: string;
  public timings: number;

  public constructor() {
    this.exportBinary = false;
    this.binaryOutputFilename = "";
    this.importBinary = false;
    this.binaryInputFilename = "";
    this.iterations = 1;
    this.radianceImageFileNameFormat = "";
    this.radianceModelFileNameFormat = "";
    this.saveModulo = 10;
    this.raytracingImageFileName = "";
    this.timings = 0;
  }
}
